spec/requests/tweets_spec.rb

```diff
@@ -0,0 +1,244 @@
+#       end
+#     end
+
+#     context "with invalid parameters" do
+#       it "does not create a new Tweet" do
+#         expect {
+#           post tweets_url, params: { tweet: invalid_attributes }
+#         }.to change(Tweet, :count).by(0)
+#       end
+
+    
+#       it "renders a response with 422 status (i.e. to display the 'new' template)" do
+#         post tweets_url, params: { tweet: invalid_attributes }
+#         expect(response).to have_http_status(:unprocessable_entity)
+#       end
+    
+#     end
+#   end
+
+#   describe "PATCH /update" do
+#     context "with valid parameters" do
+#       let(:new_attributes) {
+#         skip("Add a hash of attributes valid for your model")
+#       }
+
+#       it "updates the requested tweet" do
+#         tweet = Tweet.create! valid_attributes
+#         patch tweet_url(tweet), params: { tweet: new_attributes }
+#         tweet.reload
+#         skip("Add assertions for updated state")
+#       end
+
+#       it "redirects to the tweet" do
+#         tweet = Tweet.create! valid_attributes
+#         patch tweet_url(tweet), params: { tweet: new_attributes }
+#         tweet.reload
+#         expect(response).to redirect_to(tweet_url(tweet))
+#       end
+#     end
+
+#     context "with invalid parameters" do
+    
+#       it "renders a response with 422 status (i.e. to display the 'edit' template)" do
+#         tweet = Tweet.create! valid_attributes
+#         patch tweet_url(tweet), params: { tweet: invalid_attributes }
+#         expect(response).to have_http_status(:unprocessable_entity)
+#       end
+    
+#     end
+#   end
+
+#   describe "DELETE /destroy" do
+#     it "destroys the requested tweet" do
+#       tweet = Tweet.create! valid_attributes
+#       expect {
+#         delete tweet_url(tweet)
+#       }.to change(Tweet, :count).by(-1)
+#     end
+
+#     it "redirects to the tweets list" do
+#       tweet = Tweet.create! valid_attributes
+#       delete tweet_url(tweet)
+#       expect(response).to redirect_to(tweets_url)
+#     end
+#   end
+# end
+
+
+require 'rails_helper'
+
+RSpec.describe "users", type: :request do
+    describe "Get /stats" do
+        it "returns http success" do
+        get stats_tweet_path(:tweet)
+
+        expect(response.status).to eq(200)
+        expect(response).to match_response1_schema("tweet")
+        end  
+    end
+
+    
+    describe "Post /like" do
+        it "returns http success" do
+            tweet1 = create(:tweet)
+            user1 = create(:user)
+        post like_tweet_path(:like)  
+
+        expect(response.status).to eq(200)
+        expect(response).to match_response_schema("like")
+        end  
+    end 
+
+    describe "delete /dislike" do
+        it "returns http success" do
+         delete dislike_tweet_path(:like)
+
+        expect(response.status).to eq(200)
+        expect(response).to match_response_schema("tweet")
+        end  
+    end 
```

Comment:
Your test failed because you did not setup any previous data to test the delete action of a like

```
users
  delete /dislike
    returns http success (FAILED - 1)

Failures:

  1) users delete /dislike returns http success
     Failure/Error: like = Like.find(params[:tweet_id])
     
     ActiveRecord::RecordNotFound:
       Couldn't find Like without an ID
     # ./app/controllers/tweets_controller.rb:27:in `disliked'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/actiontext-7.0.6/lib/action_text/rendering.rb:20:in `with_renderer'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/actiontext-7.0.6/lib/action_text/engine.rb:69:in `block (4 levels) in <class:Engine>'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/tempfile_reaper.rb:15:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/etag.rb:27:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/conditional_get.rb:40:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/head.rb:12:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/session/abstract/id.rb:266:in `context'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/session/abstract/id.rb:260:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/rack/logger.rb:40:in `call_app'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/rack/logger.rb:25:in `block in call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/rack/logger.rb:25:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/method_override.rb:24:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/runtime.rb:22:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-2.2.8/lib/rack/sendfile.rb:110:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/railties-7.0.6/lib/rails/engine.rb:530:in `call'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-test-2.1.0/lib/rack/test.rb:360:in `process_request'
     # /home/dev0/.rvm/gems/ruby-3.1.2/gems/rack-test-2.1.0/lib/rack/test.rb:153:in `request'
     # ./spec/requests/tweets_spec.rb:164:in `block (3 levels) in <top (required)>'

Finished in 0.02083 seconds (files took 0.44934 seconds to load)
1 example, 1 failure
```